contracts/pool/concentrated/ConcentratedLiquidityPool.sol

```diff
@@ -123,10 +124,11 @@ contract ConcentratedLiquidityPool is IPool {
             _deployData,
             (address, address, uint24, uint160, uint24)
         );
-
+        if (TickMath.MIN_SQRT_RATIO <= price && price <= TickMath.MAX_SQRT_RATIO) revert BadPrice();
```

Comment:
This seems to revert if the price is valid?

Reply:
This would revert when the price is invalid, as the pool creator can choose whatever price they want, it should be in the specified range i.e. above MIN_SQRT_RATION and below MAX_SQRT_RATIO.

Reply:
> if (TickMath.MIN_SQRT_RATIO <= price && price <= TickMath.MAX_SQRT_RATIO) revert BadPrice();

You revert only if the price is in the specified range

Reply:
Ah, yes, just fixed it. It was a WIP but should be okay now